### PR TITLE
Upgraded map-layers packages core packages 4.0

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

**Comment by mdastous-bentley** [comment]
@eringram I want to start some 4.0.x development, can I use this branch as 4.0.x base branch?

**Comment by eringram** [comment]
> @eringram I want to start some 4.0.x development, can I use this branch as 4.0.x base branch?

Yeah that sounds good

**Comment by mdastous-bentley** [comment]
@aruniverse I assume you have to change the version flags in rush config again?  Do we need to build before merge this PR?

File: common/changes/@itwin/map-layers/eringram-maplayers-remove-internals_2023-05-05-13-21.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/map-layers",
+      "comment": "\"Updated to itwin.js core packages to latest 4.0dev version.  Solved api changes\"",
+      "type": "minor"
```
Comment (aruniverse) [inline common/changes/@itwin/map-layers/eringram-maplayers-remove-internals_2023-05-05-13-21.json:6]:
```suggestion
      "type": "none"
```

File: common/changes/@itwin/map-layers/eringram-maplayers-remove-internals_2023-05-05-13-21.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/map-layers",
+      "comment": "\"Updated to itwin.js core packages to latest 4.0dev version.  Solved api changes\"",
```
Comment (aruniverse) [inline common/changes/@itwin/map-layers/eringram-maplayers-remove-internals_2023-05-05-13-21.json:5]:
```suggestion
      "comment": "Update and react to iTwin.js Core 4.0",
```

File: packages/itwin/map-layers/src/ui/widget/SubLayersTree.tsx [hunk 3]
```diff
@@ -111,7 +110,8 @@ function getStyleMapLayerSettings(settings: ImageMapLayerSettings, isOverlay: bo
     showSubLayers: true,
     isOverlay,
     layerIndex,
-    provider: IModelApp.mapLayerFormatRegistry.createImageryProvider(settings),
+    // provider: IModelApp.mapLayerFormatRegistry.createImageryProvider(settings),
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/itwin/map-layers/src/ui/widget/SubLayersTree.tsx]: done

**Review by aruniverse — APPROVED** [review]

**Pull request merged**